### DISCUSSION TITLE
Keeping multiple active child nodes in BackStack

### DISCUSSION THREAD

**Discussion by saket** [bumble-tech/appyx#123] [Q&A]
Hey folks, I'm trying to create a navigation system where at least two nodes are active at all times in `BackStack`. This is to mostly to support screens with transparent backgrounds. Is this something that's doable right now without rolling out my own `NavModel`?

**Comment by KovalevAndrey** [comment]
Hi @saket, `BackStack` implementation of `NavModel` doesn't allow to do that but we have a separate API for that. Check out `ParentNode`'s `PermanentChild`. It allows you to keep associated `Node` in the `PermanentNavModel` alive as long as `ParenNode` is alive

**Reply by saket** [reply]
Whoops sorry I should have clarified that by I was referring to doing this dynamically. I think I'm effectively trying to keep two nodes as `ACTIVE`. Something along these lines:

```kotlin
val children by backStack.elements.collectAsState()
children.takeLast(2).forEach { element ->
  Child(routingElement = element)
}
```

Of course this code doesn't work because `BackStack` only wants to keep one active child at a time.

**Reply by KovalevAndrey** [reply]
You can try to define your own pop and push operations. In your push operation you can check the amount of elements in the list, and if there's only one you will not change its target state as STASHED and leave it as ACTIVE.

**Reply by saket** [reply]
By "defining my own operations", are you referring to implementing `BackStackOperation`?

**Reply by KovalevAndrey** [reply]
Yes, you just need your custom Backstack operations with which you will manipulate backstack elements in your own custom way. The rest will be handled by the library

**Reply by saket** [reply]
`BackStackOperation` is a sealed interface so that sounds difficult 😄

**Reply by KovalevAndrey** [reply]
Wops, I even forgot it was sealed. You could play with it in your own fork to see if defining your own operations does the job for you. In the mean time you can submit a feature request for allowing clients to define their own operations

**Reply by saket** [reply]
I can try! 

`BackStackSlider` depends on transition operations for calculating offset animations. What would this when block look like if `BackStackOperation` isn't sealed?

https://github.com/bumble-tech/appyx/blob/6c9d75d919d241b9cc496a0d9a2f50fb2d659c99/core/src/main/kotlin/com/bumble/appyx/navmodel/backstack/transitionhandler/BackStackSlider.kt#L50

**Reply by KovalevAndrey** [reply]
this means that we should add `else` branch for any other operations and define a behaviour for them. For instance, `toOutsideRight(width)`. This is not a big deal as clients can easily create their own transition handlers

**Reply by zsoltk** [reply]
@saket It's sealed only for purposes of transition handling (exhaustive when) and we were considering keeping it open. 

However we went this way, as a core concept of the library is that you can implement your own navigation models with whatever states you wish anyway.

We even had an example with a "card stack" having more than one ACTIVE state.

**Reply by KovalevAndrey** [reply]
https://github.com/bumble-tech/appyx/pull/130 sealed interface is removed
